### DISCUSSION TITLE
Image upload - server side

### DISCUSSION THREAD

**Discussion by sunnysonx** [strapi/strapi#6159] [Legacy Topics]
Hello,
Lets say I have a content-type called Products, I scrape the Product's title, description, image from another website when I add its link to strapi, on beforeSave:
```
beforeSave: async (model, attrs, options) => {
    if (model.productUrl){
      let productDetails = await getProductDetailsFromAnotherWebsite(model.productUrl);
      model.title = productDetails.data.title;
      model.description = productDetails.data.description;
      model.image = productDetails.data.image;
    }
  }
```
So here I have Image's static URL in productDetails.data.image, how do I download the image and upload it to strapi using Upload-plugin to map them correctly? I saw the documentation, it explains how to upload files from the frontend, assuming I don't have any frontend, how can I do it from the server side and map the uploaded image properly to the Product? Any ideas are welcomed, thank you.

**Comment by lauriejim** [comment]
Hello! The best way when there is no documentation like that is to check in the source code.
Strapi let you upload files with the `POST /upload` API endpoint.
So I went into the `strapi-plugin-upload` plugin code and check the controller related to this endpoint.
https://github.com/strapi/strapi/blob/ceb113a865c41a393818f2b0c6570ac3a9704ac9/packages/strapi-plugin-upload/controllers/Upload.js#L45-L46
You will be able to understand how all the process works.

**Reply by sunnysonx** [reply]
Makes sense. That way I could use upload() function from the controller to upload them.
Thank you!
